### PR TITLE
initial commit and fix to c609

### PR DESCRIPTION
**Pull request by edsu7**
See https://github.com/icgc-argo/argo-data-submission/issues/40

- fixed c609's regex; added escape character to '\'
- added support for nested cram fileMd5sum and fileSize
- added affiliated files needed for test
- added test cases `anon_chr1_cram.valid` and `anon_chr1_cram.invalid`
- updated older expected reports to reflect c685

### CODE REVIEW

File: seq_tools/validation/c609_fastq_sanity.py [hunk 1]
```diff
@@ -147,9 +145,9 @@ def fastq_test_regex(fastq,path):
             return False,"Line #%s within FASTQ file %s not following FASTQ format, missing '+' at start of the line" % (line_tracker+1,fastq)
         elif line_count%4==3 \
             and \
-            re.findall(r'[^'+''.join([chr(phred+64) for phred in range(0,41)])+']',line) \
+            re.findall('%r'%'[^'+r''.join([chr(phred+64) for phred in range(0,41)])+']',line) \
             and \
-            re.findall(r'[^'+''.join([chr(phred+33) for phred in range(0,42)])+']',line):
+            re.findall('%r'%'[^'+r''.join([chr(phred+33) for phred in range(0,42)])+']',line) :
```
Comment (lindaxiang) [inline seq_tools/validation/c609_fastq_sanity.py:150]: This returns `'[^'!"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJ]` of the pattern, is it correct?

File: seq_tools/validation/c609_fastq_sanity.py [hunk 2]
```diff
@@ -116,9 +114,9 @@ def check(self):
 def fastq_test_length(fastq,path):
     file_path=os.path.join(path,fastq)
     if fastq.endswith("fastq.gz") or fastq.endswith("fq.gz"):
-        cmd="zcat "+file_path+" | wc -l"
+        cmd="cat "+file_path+" | zcat | wc -l"
```
Comment (lindaxiang) [inline seq_tools/validation/c609_fastq_sanity.py:117]: why to make these changes?

File: seq_tools/validation/c609_fastq_sanity.py (hunk 1 above)
Comment (lindaxiang) [reply seq_tools/validation/c609_fastq_sanity.py:150]: This PR is mainly for c685. Do you mind to move fixes to c609 into separate PR?

File: seq_tools/validation/c685_nested_cram_fileMd5sum_and_fileSize_match.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    """
+
+    def __init__(self, ctx, metadata, skip=False):
+        super().__init__(
+            ctx=ctx,
+            metadata=metadata,
+            checker_name=__name__,
+            depends_on=[  # dependent checks
+                'c180_file_uniqueness',
+                'c605_all_files_accessible'
```
Comment (lindaxiang) [inline seq_tools/validation/c685_nested_cram_fileMd5sum_and_fileSize_match.py:41]: Do we need to make changes to c605 accordingly to include file accessible tests for nested `original_cram` if it does have some?

**Review by lindaxiang — CHANGES_REQUESTED** [review]

File: seq_tools/validation/c609_fastq_sanity.py (hunk 1 above)
Comment (edsu7) [reply seq_tools/validation/c609_fastq_sanity.py:150]: https://github.com/icgc-argo/seq-tools/pull/99#discussion_r997424515
Should be:
`'[^!"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJ]'` (missing backslash).
The change was b/c the formatting would result in the `\` being treated as an escape character and not looking for `\`. I tried doing `\\` but that didn't work either. Testable via:
```
>>>line='AGGH\!'
>>> re.findall(r'[^'+''.join([chr(phred+33) for phred in range(0,42)])+']',line)
['\\']
>>> re.findall('%r'%'[^'+r''.join([chr(phred+33) for phred in range(0,42)])+']',line)
[]
```
https://github.com/icgc-argo/seq-tools/pull/99#discussion_r997426308
Sure. Sorry was lazy

File: seq_tools/validation/c609_fastq_sanity.py (hunk 2 above)
Comment (edsu7) [reply seq_tools/validation/c609_fastq_sanity.py:117]: This was a weird issue that came up during testing regarding mac using `zcat`:
https://serverfault.com/questions/570024/zcat-gzcat-works-in-linux-not-on-osx-general-linux-osx-compatibility

**Comment by edsu7** [comment]
Created PRs as per Linda's suggestionL
https://github.com/icgc-argo/seq-tools/pull/100
https://github.com/icgc-argo/seq-tools/pull/101

**Pull request merged**